### ISSUE TITLE
The API checker should report private field changes as ABI breaks only

### ISSUE THREAD

**Issue by gilles-peskine-arm** [Mbed-TLS/mbedtls#6592]
`abi_check.py` reports changes in structures defined in `include/*/*.h` as API breaks. But for the most part, they aren't: we declare most fields as private, and changes to private fields are only ABI changes, not API breaks.

The goal of this task is to change `abi_check.py` to report changes in private fields in public structures as only ABI changes and not API changes.

**Comment by mpg** [comment]
@gilles-peskine-arm Is there a difference between this and #4374?

**Comment by gilles-peskine-arm** [comment]
@mpg No, thanks for pointing it out. I guess I forgot to search, or maybe I only searched for `abi_check` which is misspelled in the earlier issue.
